packages/node/test/requestdata.test.ts

```diff
@@ -0,0 +1,487 @@
+/* eslint-disable deprecation/deprecation */
+
+/* Note: These tests (except for the ones related to cookies) should eventually live in `@sentry/utils`, and can be
+ * moved there once the the backwards-compatibility-preserving wrappers in `handlers.ts` are removed.
```

Comment:
Why can't we just duplicate some of these tests?

Reply:
Because of the sunk cost fallacy and plan continuation bias?

In other words, no especially good reason, but at this point I'd rather move on and come back to it when we do the v8 deprecations.

Here's what happened: In the original version of this PR, the function signatures were the same between the old and new versions of the functions, so it was easy to just run the tests twice, once for each version, new and old. Then I added the DI, the signatures changed, and I thought, _Oh, I'll just add a quick helper to mush them into being the same. Shouldn't be that hard, and besides, everyone knows that having two copies of things is bad..._ But in the end the mushing was uglier and more involved than I'd hoped, and you're right in your implication that the fact that we're testing stuff which is going to be deprecated makes maintenance (and therefore duplication) much less of an issue, but... it works. It's ugly, but it's temporary, and I volunteer to be the one to disentangle everything when the time comes.

So unless you feel strongly about it, I'd like to leave as Future Katie's problem.

Reply:
That sounds reasonable enough to me, let's carry forward with this then!